auditlog/models/rule.py

```diff
@@ -227,12 +228,25 @@ def create_full(self, vals, **kwargs):
         def create_fast(self, vals, **kwargs):
             self = self.with_context(auditlog_disabled=True)
             rule_model = self.env['auditlog.rule']
-            vals2 = dict(vals)
```

Comment:
Hi @mtelahun 

Thanks for the PR!
Instead of generating a full log when we get a list of dictionaries, what about this:
- replace the above line by making a deep copy of `vals`: `vals2 = copy.deepcopy(vals)`
- as the `create` method keep the order (record IDs should fit the order of dictionaries provided), set `new_values` with something like:
```python
for new_record_id, record_vals in zip(new_record.ids, vals2):
    new_values = {new_record_id: vals2}
    rule_model.sudo().create_logs(
        self.env.uid, self._name, [new_record_id],
        'create', None, new_values, {'log_type': log_type})
````

What do you think?

Reply:
Hi @sebalix 
I thought about that, but decided against it because there is no guarantee (at least in the ORM reference) that the returned recordset is in the same order as the list passed to the create() method. Specifically, I wasn't sure what would happen if the '_order' field of the model specified an ordering other than the 'id' of the record. Given that this is supposed to be an auditing tool I thought it would be better to opt for correctness rather than performance.

Reply:
I understand, and before proposing that I read a bit the code in `odoo/models.py` and the assumption that the order is the same is right (unless I misread the code). But I didn't test it myself... the `_order` should occur only on a search operation, here we are more in a `browse` usage where the order of IDs and recordset returned should be the same. To confirm.

The fast log was created to reduce the performance penalty of the full log, and I don't know how much the create multi is used by Odoo v12, but it could lead to big issues from a UX experience depending on the data model.